File: my-vue-app/src/api/NhanVienService.js
import axiosInstance from "./axios";

const BASE_API = "api/nhanvien/";

const handleError = (error) => {
  console.error("API call error:", error);
  throw error;
};

const NhanVienApi = {
  getAllNhanVien() {
    return axiosInstance.get(`${BASE_API}getAllNhanVien`).catch(handleError);
  },
  getNhanVienID(id) {
    return axiosInstance
      .get(`${BASE_API}getNhanVienID?Id_NV=${id}`)
      .catch(handleError);
  },
  getAllChucVu_IdPB(id) {
    return axiosInstance
      .get(`${BASE_API}getAllChucVu_IdPB?Id_PB=${id}`)
      .catch(handleError);
  },
  createNhanVien(data) {
    return axiosInstance
      .post(`${BASE_API}CreateNhanVien`, data)
      .catch(handleError);
  },
  updateNhanVien(data) {
    return axiosInstance
      .put(`${BASE_API}UpdateNhanVien`, data)
      .catch(handleError);
  },
  deleteNhanVien(id) {
    return axiosInstance
      .delete(`${BASE_API}DeleteNhanVien?id=${id}`)
      .catch(handleError);
  },
};

export default NhanVienApi;

File: Demo_API_QL_NS/Demo_API_QL_NS/Services/NhanVien/NhanVienService.cs
using Demo_API_QL_NS.Models;
using Demo_API_QL_NS.Repositories.NhanVien;
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Data;
using System.Threading.Tasks;
using System.Collections;

namespace Demo_API_QL_NS.Services.NhanVien
{
    public class NhanVienService : INhanVienService
    {
        private INhanVienRepository _reposiory;

        public NhanVienService(INhanVienRepository NhanVienRepository)
        {
            _reposiory = NhanVienRepository;
        }
        public async Task<IEnumerable<NhanVienDTO>> getAllNhanVien()
        {
            return await _reposiory.getAllNhanVien();

        }
        
        public async Task<bool> CreateNhanVien(NhanVienDTO model)
        {
            return await _reposiory.CreateNhanVien(model);
        }
        public async Task<bool> UpdateNhanVien(NhanVienDTO model)
        {
            return await _reposiory.UpdateNhanVien(model) ;
        }
        public async Task<bool> DeleteNhanVien(int id)
        {
            return await _reposiory.DeleteNhanVien(id);
        }
        public async Task<bool> DeleteNhanViens(List<int> ids)
        {
            return await _reposiory.DeleteNhanViens(ids);
        }
        public async Task<int> CheckId_PB_NhanVien(int Id_PB)
        {
            return await _reposiory.CheckId_PB_NhanVien(Id_PB);
        }
        public async Task<int> CheckId_CV_NhanVien(int Id_CV)
        {
            return await _reposiory.CheckId_CV_NhanVien(Id_CV);
        }
        public async Task<IEnumerable<NhanVienDTO>> SearchNhanVienHT(string HoTen)
        {
            return await _reposiory.SearchNhanVienHT(HoTen);
        }
        public async Task<NhanVienDTO> getNhanVienID(int Id_NV)
        {
            return await _reposiory.getNhanVienID(Id_NV);
        }
    }
}
